docs/api.md

```diff
@@ -53,13 +53,94 @@ In our experience, any system that is successful needs to grow and change as new
+in interfaces based on your feedback. In later stages, changes may not be possible.** Since alpha
+features are under active development, you may wish to try alpha features in a test cluster, rather
+than in a production cluster.
+
+Different API versions imply different levels of stability and support:
+ - Alpha level:
+   - The version names contain `alpha` (e.g. `v1alpha1`).
+   - *Purpose*: sharing new features with developers and eager expert users, prior to:
+     - *complete implementation*: for instance, some API operations, CLI
+     commands, or UI support may not be implemented. 
+     - *thorough API review*: the object schema may change in a later software release, and it may be necessary
+       to delete/modify/re-create objects in order to upgrade a cluster to a subsequent software release.
+       The object modification may not be easily automated, and the process may be poorly documented.
+     - *commitment to complete the feature*: the feature may be dropped entirely in a later software release.
+     - *allow rapid changes in versions*: API versions may increment faster than the minor release cadence.
+   - Alpha features are present in released binaries, but disabled by default. Enabling an Alpha feature
+     it is be necessary to make flag changes on the API server, controller, kubelet, etc..
+     Hosted clusters (e.g. GKE) may not allow users to enable alpha object versions and features at all.
+  
+ - Beta level:
+   - The version names contain `beta` (e.g. `v2beta3`).
+   - *Purpose*: getting feedback from a wider set of users on usability and API consistency, while offering
+     enough stability to allow use in clusters running realistic loads (e.g. QA cluster for a large web service).
+   - Before graduating from alpha to beta:
+     - a feature is fully implemented; all relevant REST endpoints and kubectl commands work.  The code is tested to high standards.
+     - a feature has had an API review, though the beta period is still likely to turn up API issues not
+       found in review.
+     - the project is committed to graduating the feature to stable (typically within 3 months, but sometimes longer)
+     - has been tested to high standards and is considered safe to enable in a production cluster.
+   - The object schema may change in a later software release.  When this happens, an automatic
+     upgrade of a cluster with beta objects may not be possible.  Instead, it may be necessary
+     to delete/modify/re-create objects in order to upgrade a cluster to the new schema.  When this happens,
+     for a beta feature, we will provide documentation on how to do this.
+   - Releases will support two consecutive versions (e.g. `v1beta1` and `v1beta2`; or `v1beta2` and `v1`)
+     for at least one minor release cycle (typically 3 months) so that users have enough time
+     to upgrade and migrate objects.
+   - TBD: do beta features enable by default?
```

Comment:
Yes beta resources (HPA, Job and ingress) should be on by default.
Users should be able to enable/disable each extension resource. 
https://github.com/kubernetes/kubernetes/pull/15516